src/tables.jl

```diff
@@ -47,3 +47,587 @@ function schema_names(featuredefn::IFeatureDefnView)
+    T::Type{U},
+) where {U<:GeoInterface.AbstractGeometry}
+    return :(convert(OGRwkbGeometryType, T))
+end
+@generated function _convert_cleantype_to_AGtype(T::Type{U}) where {U}
+    return :(convert(OGRFieldType, T), convert(OGRFieldSubType, T))
+end
+
+"""
+    _convert_coltype_to_cleantype(T)
+
+Convert a table column type to a "clean" type:
+- Unions are flattened
+- Missing and Nothing are dropped
+- Resulting mixed types are approximated by their tightest common supertype
+
+"""
+function _convert_coltype_to_cleantype(T::Type)
+    flattened_T = Base.uniontypes(T)
+    clean_flattened_T = filter(t -> t ∉ [Missing, Nothing], flattened_T)
+    return promote_type(clean_flattened_T...)
+end
+
+function _create_empty_layer_from_AGtypes(
+    colnames::NTuple{N,String},
+    AGtypes::Vector{
+        Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}},
+    },
+    name::String,
+) where {N}
+    # Split names and types: between geometry type columns and field type columns
+    geomindices = isa.(AGtypes, OGRwkbGeometryType)
+    !any(geomindices) && error("No column convertible to geometry")
+    geomtypes = AGtypes[geomindices] # TODO consider to use a view
+    geomnames = colnames[geomindices]
+
+    fieldindices = isa.(AGtypes, Tuple{OGRFieldType,OGRFieldSubType})
+    fieldtypes = AGtypes[fieldindices] # TODO consider to use a view
+    fieldnames = colnames[fieldindices]
+
+    # Create layer
+    layer = createlayer(name = name, geom = first(geomtypes))
+    # TODO: create setname! for IGeomFieldDefnView. Probably needs first to fix issue #215
+    # TODO: "Model and handle relationships between GDAL objects systematically"
+    GDAL.ogr_gfld_setname(
+        getgeomdefn(layerdefn(layer), 0).ptr,
+        first(geomnames),
+    )
+
+    # Create FeatureDefn
+    if length(geomtypes) >= 2
+        for (geomtype, geomname) in zip(geomtypes[2:end], geomnames[2:end])
+            creategeomdefn(geomname, geomtype) do geomfielddefn
+                return addgeomdefn!(layer, geomfielddefn) # TODO check if necessary/interesting to set approx=true
+            end
+        end
+    end
+    for (fieldname, (fieldtype, fieldsubtype)) in zip(fieldnames, fieldtypes)
+        createfielddefn(fieldname, fieldtype) do fielddefn
+            setsubtype!(fielddefn, fieldsubtype)
+            return addfielddefn!(layer, fielddefn)
+        end
+    end
+
+    return layer, geomindices, fieldindices
+end
+
+"""
+    _infergeometryorfieldtypes(sch, rows, spgeomcols, spfieldtypes)
+
+Infer ArchGDAL field and geometry types from schema, `rows`' values (for WKT/WKB cases) and `geomcols` and `fieldtypes` kwargs 
+
+"""
+function _infergeometryorfieldtypes(
+    sch::Tables.Schema{names,types},
+    rows,
+    spgeomcols::Union{Nothing,Vector{String},Vector{Int}},
+    spfieldtypes::Union{
+        Nothing,
+        Dict{Int,Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}}},
+    },
+) where {names,types}
```

Comment:
Can we start with a simple implementation where this is
```julia
_infergeometryorfieldtypes(sch::Tables.Schema{names,types}, rows) where {names,types}
```
?

The behavior will be based on the current implementation with `spgeomcols=nothing` and `spfieldtypes=nothing`.

Let's introduce the functionality for `spgeomcols` in a follow-up PR (we can drop `spfieldtypes` in that PR per your recommendation).

Reply:
Dropped `spgeomcols` in commit e8560e4.
I have duplicated the branch before the drop, to be able to make an issue/PR, once the current PR is merged